engine/language-server/src/main/scala/org/enso/languageserver/effect/Exec.scala

```diff
@@ -47,55 +46,66 @@ case class ZioExec(runtime: effect.Runtime) extends Exec[ZioExec.IO] {
   /** Execute Zio effect.
     *
     * @param op effect to execute
+    * @param trace object prevents the library from messing up the user's execution trace
     * @return a future containing either a failure or a result
     */
-  override def exec[E, A](op: ZIO[ZEnv, E, A]): Future[Either[E, A]] = {
-    val promise = Promise[Either[E, A]]()
-    runtime.instance.unsafeRunAsync(op) {
-      _.fold(
-        ZioExec.completeFailure(promise, _),
-        ZioExec.completeSuccess(promise, _)
+  override def exec[E, A](
+    op: ZIO[ZAny, E, A]
+  )(implicit trace: Trace): Future[Either[E, A]] =
+    zio.Unsafe.unsafe { implicit unsafe =>
+      val promise = Promise[Either[E, A]]()
+      runtime.instance.unsafe.fork(
```

Comment:
> have to spawn a fiber every time we call fork on an effect to spawn off a new fiber.

OK, so now we create new thread pool, right? Previously we just `unsafe`(!)`RunAsync` - e.g. sent the execution to an existing thread pool.

Reply:
The logic didn't change, just the API. Fiber is like a green thread. All the computations are still happening in the thread pool configured on the startup.

Reply:
There are no "green threads" in the JVM we are using. [Project Loom](https://openjdk.org/projects/loom/) has only become an experimental feature of JDK 19 - without it there is no chance to have any _green threads_ in JVM.

Reply:
Correct. That's why ZIO has its own primitive called [`Fiber`](https://zio.dev/reference/fiber/).